### ISSUE TITLE
How to implement the scheduler under non-mmsegmentation environment?

### ISSUE THREAD

**Issue by ChengYuChuan** [open-mmlab/mmsegmentation#1725]
I am trying to pick a part of mmsegmentation into my original script.
here is the model I use:
https://github.com/open-mmlab/mmsegmentation/blob/master/configs/hrnet/fcn_hr48_512x1024_160k_cityscapes.py
here are the schedules within this model: https://github.com/openmmlab/mmsegmentation/blob/master/configs/_base_/schedules/schedule_160k.py
```
# optimizer
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0005)
optimizer_config = dict()
# learning policy
lr_config = dict(policy='poly', power=0.9, min_lr=1e-4, by_epoch=False)
# runtime settings
runner = dict(type='IterBasedRunner', max_iters=160000)
checkpoint_config = dict(by_epoch=False, interval=16000)
evaluation = dict(interval=16000, metric='mIoU', pre_eval=True)
```
But, I encounter some questions:
```
class PolyLrUpdaterHook(LrUpdaterHook):

    def __init__(self,
                 power: float = 1.,
                 min_lr: float = 0.,
                 **kwargs) -> None:
        self.power = power
        self.min_lr = min_lr
        super().__init__(**kwargs)

    def get_lr(self, runner: 'runner.BaseRunner', base_lr: float):
        if self.by_epoch:
            progress = runner.epoch
            max_progress = runner.max_epochs
        else:
            progress = runner.iter
            max_progress = runner.max_iters
        coeff = (1 - progress / max_progress)**self.power
        return (base_lr - self.min_lr) * coeff + self.min_lr
 ```
1. What are base_lr and self.min_lr?(if the config is like the first code part)
I think base_lr=0.01 and self.min_lr = 0.0001 (there is other parameters: max_progress = 160000, self.power=0.9)
2. how can I reproduce the scheduler without mmsegmentation env?
here is my thought: (but it didn't work)
```
args.batch_size = 5
IterationPerEpoch = float(2975 / args.batch_size)
args.num_epochs = 270
MaxIteration = args.num_epochs*IterationPerEpoch
args.lr = 0.01


for epoch in range(start_epoch, args.num_epochs+1):
  for step, (images, labels) in enumerate(loader):
    OverallIterations = (epoch - 1) * IterationPerEpoch + step
    lambda1 = lambda OverallIterations: pow((1 - OverallIterations / MaxIteration), 0.9)
    scheduler = lr_scheduler.LambdaLR(optimizer, lr_lambda=lambda1)
    scheduler.step(OverallIterations)
    optimizer.step()
```

**Comment by MeowZheng** [comment]
1. `base_lr` is initialized lr from optimizer, ref:
https://github.com/open-mmlab/mmcv/blob/2708fac6c5ec017d384a4f41fa74d91c9e79b47d/mmcv/runner/hooks/lr_updater.py#L112-L127
`min_lr` is the minimal end learning rate that is initialized when creating `PolyLrUpdaterHook` instance.

2. It might be hard to reproduce without MMSeg, as the lr scheduler is a hook of the training runner in MMCV and the training runner works for managing the whole process.

**Comment by ChengYuChuan** [comment]
I mean I want to reproduce this function get_lr from class PolyLrUpdaterHook.
in the example, I try to use lr_scheduler.LambdaLR to do the same way with the function get_lr.

Could you check it for me if it is correct?
especially this line"
```
 lambda1 = lambda OverallIterations: pow((1 - OverallIterations / MaxIteration), 0.9)
```
thank you so much

**Comment by MeowZheng** [comment]
I think you might test if it works as you expect, and I'm not quite sure what you expect.

**Comment by ChengYuChuan** [comment]
Yup, I got it right. thank you so much.
